engine/engine.go

```diff
@@ -212,7 +231,19 @@ func (e *Engine) Publish(ctx context.Context, adv schema.Advertisement) (cid.Cid
 
 	log.Infow("Publishing advertisement in pubsub channel", "cid", c.String())
 	// Use legPublisher to publish the advertisement.
-	return c, e.publisher.UpdateRoot(ctx, c)
+	err = e.publisher.UpdateRoot(ctx, c)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	if e.httpPublisher != nil {
```

Comment:
can we have the publisher be one or the other, rather than both a graphsync and http publisher?

Reply:
done